executor/runtime/types/types.go

```diff
@@ -375,14 +375,19 @@ func GenerateV1TestPod(taskID string, resources *Resources, cfg *config.Config)
 // Someday when network mode is set across the board, we can drop this function and just fail
 // fast when network mode is Unknown, but till then, this function encpasulates the busines logic
 // of interpretting the legacy attributes and computing what the effective network mode "should" be
-func computeEffectiveNetworkMode(originalNetworkMode string, assignIPv6Address bool) string {
+// Caveat: If TSA has been disabled for operational reasons, there is no provider for the Ipv6AndIpv4Fallback
+//         network mode, so return Ipv6AndIpv4 instead
+func computeEffectiveNetworkMode(originalNetworkMode string, assignIPv6Address bool, disableTsa bool) string {
 	if originalNetworkMode == titus.NetworkConfiguration_UnknownNetworkMode.String() {
 		if assignIPv6Address {
 			return titus.NetworkConfiguration_Ipv6AndIpv4.String()
 		}
 		return titus.NetworkConfiguration_Ipv4Only.String()
 	}
 	if originalNetworkMode == titus.NetworkConfiguration_HighScale.String() {
+		if disableTsa {
```

Comment:
This is not enough, you also need to catch the case where the originalNetworkMode was the transition mode.

If the transition mode is not enabled, then return dual stack

Reply:
Done.